WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -301,6 +300,68 @@ private void updateActionModeTitle() {
         }
     }
 
+    private void setupSearchView(Menu menu) {
+        MenuItem menuSearch = menu.findItem(R.id.menu_search);
+
+        mSearchView = (SearchView) menuSearch.getActionView();
+        mSearchView.setIconifiedByDefault(false);
+        mSearchView.setOnQueryTextListener(this);
+
+        MenuItemCompat.setOnActionExpandListener(menuSearch, new MenuItemCompat.OnActionExpandListener() {
+            @Override
+            public boolean onMenuItemActionExpand(MenuItem item) {
+                enableSearchMode();
+                return true;
+            }
+
+            @Override
+            public boolean onMenuItemActionCollapse(MenuItem item) {
+                disableSearchMode();
+                return true;
+            }
+        });
+
+        ifInSearchModeExpandActionViewAndSetQueryToLastSearch(menuSearch);
+    }
+
+    private void ifInSearchModeExpandActionViewAndSetQueryToLastSearch(MenuItem menuSearch) {
```

Comment:
That's quite a long routine name! :) Maybe something simpler, like `setQueryIfInSearch`? Or get rid of it altogether and move the code inline with the previous routine, since that's the only place where this is called.